### PR TITLE
Detect ARM systems for kinesis-mock installation

### PR DESCRIPTION
**Pull request by etspaceman**
An [issue](https://github.com/etspaceman/kinesis-mock/issues/128) was raised in Kinesis Mock in which M1 Macbooks were getting a segfault. This is due to the binary file for MacOS being compiled for AMD64.

This PR will detect ARM machines and ensure that they fallback to the JAR file. 

Unfortunately Kinesis Mock will not be able to build M1 compliant binary executables until GitHub Actions supports a virtual environment for this, as GraalVM requires this for linking the libraries properly:

https://github.com/actions/virtual-environments/issues/2187

### CODE REVIEW

**Review by whummer — APPROVED** [review]
LGTM, thanks for the quick fix @etspaceman ! (Awaiting review from @thrau as well.) Thanks

File: localstack/services/install.py [hunk 1]
```diff
@@ -172,9 +172,11 @@ def install_kinesis_mock():
 
     machine = platform.machine().lower()
     system = platform.system().lower()
+    version = platform.version().lower()
 
     LOG.debug('getting kinesis-mock for %s %s', system, machine)
-    if machine == 'x86_64' or machine == 'amd64':
+    if ((machine == 'x86_64' or machine == 'amd64') and
+        "arm64" not in version and "arm32" not in version):
```
Comment (thrau) [inline localstack/services/install.py:179]: could we maybe make the check a bit more explicit, or at least add a comment, that this is a workaround for M1?

something along the lines of
`is_probably_m1 = system == 'darwin' and ('arm64' in version or 'arm32' in version)`

Comment (thrau) [inline localstack/services/install.py:179]: we should stick consistently with single quotes for strings `'`

**Review by thrau — APPROVED** [review]
The logic seems fine to me. we have this as an example output from an M1 machine:

```
python3 -c 'import platform; print(platform.uname());

uname_result(system='Darwin', node='m1book', release='20.4.0', version='Darwin Kernel Version 20.4.0: Thu Apr 22 21:46:41 PDT 2021; root:xnu-7195.101.2~1/RELEASE_ARM64_T8101', machine='arm64', processor='arm')
```

vis a vis a normal mac book

```
$ python3 -c 'import platform; print(platform.uname())';
uname_result(system='Darwin', node='macbookpro', release='19.6.0', version='Darwin Kernel Version 19.6.0: Mon Aug 31 22:12:52 PDT 2020; root:xnu-6153.141.2~1/RELEASE_X86_64', machine='x86_64', processor='i386')
```

i just have a couple of comments about the formatting, nothing major.

and thanks for being committed to making your integration work :-)

File: localstack/services/install.py (hunk 1 above)
Comment (whummer) [reply localstack/services/install.py:179]: This is actually an issue in the lint setup - I'll fix that in a follow-up PR, no need to change here.. 👍 (but thanks for the pointer!)

Comment (etspaceman) [reply localstack/services/install.py]: Updated

**Pull request merged**